### ISSUE TITLE
Sort Desc for DataTables is not working

### ISSUE THREAD

**Issue by rafaehlers** [gravityview/GravityView#1271]
![gv-sort-desc](https://user-images.githubusercontent.com/1222472/51569019-5872a180-1e82-11e9-8604-57de4461cc49.png)

Tested on GV 2.2.4 and DT 2.3.4

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/995529792029955/1100530894678736)

**Comment by rafaehlers** [comment]
https://secure.helpscout.net/conversation/676829849/16986/

**Comment by rafaehlers** [comment]
https://secure.helpscout.net/conversation/768964068/18967

**Comment by soulseekah** [comment]
This should be fixed real good with the latest DT `develop` multi-sort fixes, etc. Please confirm?

https://github.com/gravityview/DataTables/archive/develop.zip

**Comment by rafaehlers** [comment]
https://secure.helpscout.net/conversation/828020567/20219/

**Comment by soulseekah** [comment]
Does it not work? Probably fixes #1299 as well?

**Comment by rafaehlers** [comment]
https://secure.helpscout.net/conversation/864992651/20889

**Comment by rafaehlers** [comment]
https://secure.helpscout.net/conversation/886478096/21268/

**Comment by soulseekah** [comment]
This seems to work very well for me. I've taken a lot of time to test it.

However, I think I know what might be going on, and I'll need you to check, @rafaehlers 
DataTables caches the selected column order so after refreshing the page it takes the configuration from Session Storage instead of the dtConfig that we pass it.

This is the `stateSave`, we use it actively to make sure that DT stays on the same internal page when reloading the page itself (unless it's a search).

So this issue might be a confusion with previous saved states where users clicked the sort columns for testing purposes or something.

**Comment by rafaehlers** [comment]
@soulseekah you're right! Session Storage keeps the column ID and the sort order from the last time sorting on the table was used, that's why refreshing the page doesn't make it sort by the View Settings. What are the next steps now?

**Comment by rafaehlers** [comment]
https://secure.helpscout.net/conversation/889267164/21321

**Comment by rafaehlers** [comment]
https://secure.helpscout.net/conversation/892030810/21358/

**Comment by soulseekah** [comment]
Turning off the state caching functionality will help with all of these. But will introduce an issue where a refresh of the page will reset the view back to page 1 with the set sorting. I highly doubt anyone will ever want this tradeoff.

The way to do this is via the `gravityview_datatables_js_options` filter.
They can set `stateSave` to false in the array.

Hope this helps, @rafaehlers

**Comment by rafaehlers** [comment]
https://secure.helpscout.net/conversation/894492094/21392

**Comment by rafaehlers** [comment]
https://secure.helpscout.net/conversation/899523548/21487/

**Comment by rafaehlers** [comment]
Leaving this here: https://gist.github.com/rafaehlers/2182c05594772af245db630047646428

**Comment by rafaehlers** [comment]
```
add_filter( 'gravityview_datatables_js_options', function( $dt_config, $view_id, $post ) {
    $dt_config['order'] = array();
    $dt_config['stateSave'] = false;
    return $dt_config;
}, 10, 3 );```

Works!

**Comment by zackkatz** [comment]
> ```
> add_filter( 'gravityview_datatables_js_options', function( $dt_config, $view_id, $post ) {
>     $dt_config['order'] = array();
>     $dt_config['stateSave'] = false;
>     return $dt_config;
> }, 10, 3 );
> ```
> 
> 
> Works!

@jlgrisham Can you please convert this into a doc?

**Comment by rafaehlers** [comment]
This is not working again. My code above doesn't work as well: https://github.com/gravityview/GravityView/issues/1271#issuecomment-515207108

**Comment by bennemann** [comment]
https://secure.helpscout.net/conversation/11(1630420/26072

**Comment by rafaehlers** [comment]
Fixed with DT 2.4.3 and GV 2.8.1
